### PR TITLE
[SUPERCEDED] Initial port from UBports

### PR DESCRIPTION
**Pull request by tari01**
I have tested it by running it with blueman-applet side-by-side. Toggling the switches work as expected - the actions fire, and bluez reacts properly.

The problem is with incoming bluez events:
```
bluetooth.notify["enabled"].connect
bluetooth.notify["connected"].connect
```
The handlers for these never get called, so the indicator does not update.

### CODE REVIEW

**Review by sunweaver — APPROVED** [review]
LGTM, waiting for one more feedback from someone at UBports, e.g. @mariogrip or @dobey

File: src/desktop.vala
Comment (mariogrip) [inline src/desktop.vala:111]: Why did this get removed?

File: src/profile.vala
Comment (mariogrip) [inline src/profile.vala:129]: Explain this?

Why did action.activate.connect get replaced with action.change_state.connect functions and action.change_state.connect gets completely removed

File: src/desktop.vala
Comment (mariogrip) [inline src/desktop.vala:258]: Same here 

Why did action.activate.connect get replaced with action.change_state.connect functions and action.change_state.connect gets completely removed

File: src/desktop.vala
Comment (mariogrip) [reply src/desktop.vala:111]: Why did a.activate.connect get replaced with a.notify["state"].connect functions and a.notify["state"].connect gets completely removed

Comment (mariogrip) [reply src/desktop.vala:111]: I assume a.set_state set the state then a.notify["state"] will pick that up, but we need to set state dont we?

**Review by sunweaver — CHANGES_REQUESTED** [review]
@tari01 I pull back my LGTM. We should really split this massive PR at least into two or three PRs (or at least commits). This makes more work, but it provides a cleaner approach if we ever need to hunt bugs around the commits where this change got introduced.

  * PR 1: ubuntu system indicator -> ayatana system indicator
  * PR 2: automake -> cmake change-over
  * PR 3: state change notification modifications

Regarding the state change notifications, I am aware of the PRs for Ayatana IDO that these changes here depend on. However, how do these changes play together with qmenumodel? Have you tested that?

File: src/desktop.vala
Comment (tari01) [reply src/desktop.vala:111]: This old approach sometimes fails to get the right state after activation. There is now a much more elegant and client-friendly way for IDO:

After activation (when the state has changed), the owner of the action receives the "activate" signal with the new state as a parameter. No need to subscribe to the state change or request a state change manually any more.

File: src/profile.vala
Comment (tari01) [reply src/profile.vala:129]: > Explain this?
> 
> Why did action.activate.connect get replaced with action.change_state.connect functions and action.change_state.connect gets completely removed

Please see https://github.com/AyatanaIndicators/ayatana-indicator-bluetooth/pull/1#discussion_r496245536

File: src/desktop.vala
Comment (tari01) [reply src/desktop.vala:258]: > Same here
> 
> Why did action.activate.connect get replaced with action.change_state.connect functions and action.change_state.connect gets completely removed

Please see https://github.com/AyatanaIndicators/ayatana-indicator-bluetooth/pull/1#discussion_r496245536

**Comment by tari01** [comment]
> Regarding the state change notifications, I am aware of the PRs for Ayatana IDO that these changes here depend on. However, how do these changes play together with qmenumodel? Have you tested that?

The changes in IDO contain no hacking, only contributes to a simplified approach (plus the old one was buggy in GTK). IDO sends a standard action message with a param type specified by the client, so there is nothing qmenumodel should not understand.

**Comment by tari01** [comment]
> We should really split this massive PR at least into two or three PRs (or at least commits). This makes more work, but it provides a cleaner approach if we ever need to hunt bugs around the commits where this change got introduced.
> 
>     * PR 1: ubuntu system indicator -> ayatana system indicator
> 
>     * PR 2: automake -> cmake change-over
> 
>     * PR 3: state change notification modifications

The first two suggested commits interdepend, so they needed to stay together. I have singled out the files affected by the code change as a separate commit.

File: update-po.sh
Comment (dobey) [inline update-po.sh:1]: Why do we need this script and the `update-pot.sh` script, rather than simply using the `Intltool.cmake` functions in `po/CMakeLists.txt` instead?

File: debian/source/format
Comment (dobey) [inline debian/source/format:1]: Why not `3.0 (native)` instead?

File: data/ayatana-indicator-bluetooth.service.in
Comment (dobey) [inline data/ayatana-indicator-bluetooth.service.in:1]: Why weren't these moved with `git mv` and then modified, so that it shows up as rename with only the necessary changes in the diff? It makes some of these changes harder to follow, so better practice is to use `git mv` to rename the files then edit them, when this is necessary.

File: update-po.sh
Comment (tari01) [reply update-po.sh:1]: I was following the established model for all Ayatana Indicators using CMake:
https://github.com/AyatanaIndicators/ayatana-indicator-session/blob/master/update-po.shhttps://github.com/AyatanaIndicators/ayatana-indicator-datetime/blob/master/update-po.sh

File: debian/source/format
Comment (tari01) [reply debian/source/format:1]: @sunweaver might have an answer to this one

File: data/ayatana-indicator-bluetooth.service.in
Comment (tari01) [reply data/ayatana-indicator-bluetooth.service.in:1]: This was not intended to be a pull request, but a direct merge addressing the move to cmake and the switch to Ayatana namespace, where all the issues would be ironed out afterwards (such as the bug we inherited from upstream, mentioned in the comment).

**Comment by sunweaver** [comment]
On  Fr 02 Okt 2020 21:11:44 CEST, Robert Tari wrote:

> @tari01 commented on this pull request.
>
>
>
>> @@ -0,0 +1,11 @@
> +[Unit]
>
> This was not intended to be a pull request, but a direct merge  
> addressing the move to cmake and the switch to Ayatana namespace,  
> where all the issues would be ironed out afterwards (such as the bug  
> we inherited from upstream, mentioned in the comment).

@tari01: @Rodney: I plan to go over this massive change and break it  
up into several commits. With the woe of porting to Ayatana Indicators  
first, with Makefile.am changes and then port from automake to cmake.

 From my experience "git mv" is not needed nowadays anymore, Git is  
intelligent enough to recognize file renamings + changes (if not above  
a certain %-threshold). I'll make sure, the renamings get tracked when  
redoing that one-shot commit.

Greets,
Mike
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler Str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

File: update-po.sh
Comment (dobey) [reply update-po.sh:1]: Huh. Can you answer this question of mine @sunweaver ?

**Comment by sunweaver** [comment]
On  Fr 02 Okt 2020 21:50:55 CEST, Rodney wrote:

> @dobey commented on this pull request.
>
>
>
>> @@ -0,0 +1,40 @@
> +#!/bin/bash
>
> Huh. Can you answer this question of mine @sunweaver ?

At the time of bringing all the translation stuff to  
hosted.weblate.org, I wasn't aware of the Intltool.cmake thingy.

What would be the workflow for updating .pot files with cmake, then? I  
am happy to port that over (while keeping the shall scripts for my  
personal convenience).

Greets,
Mike
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler Str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by dobey** [comment]
At the time of bringing all the translation stuff to hosted.weblate.org, I wasn't aware of the Intltool.cmake thingy. What would be the workflow for updating .pot files with cmake, then? I am happy to port that over (while keeping the shall scripts for my personal convenience). Greets, Mike

The `po/CMakeLists.txt` would look something like https://github.com/ubports/indicator-power/blob/xenial/po/CMakeLists.txt

And then when you need to update the pot file because you changed a string in your branch, just do `make indicator-power.pot` in the build dir for example.

**Comment by tari01** [comment]
> @tari01: @rodney: I plan to go over this massive change and break it up into several commits. With the woe of porting to Ayatana Indicators first, with Makefile.am changes and then port from automake to cmake.

45 files changed in one commit is a total mess. If I had known we were going to have this audited by UBports before the merge, I would not have made all the changes in one go.

@sunweaver 
I can push the final product for this to /tari01 if you need it. Also note, the IDO switch has an annoying bug fixed in  https://github.com/AyatanaIndicators/ayatana-ido/pull/16

**Comment by sunweaver** [comment]
Superceded by PRs #2 - #11. Thus, closing...

**Pull request closed without merging**
